vital/set.h

```diff
@@ -94,6 +94,9 @@ class set
    */
   virtual T at( size_t index ) = 0;
   virtual T const at( size_t index ) const = 0;
+
+  virtual T operator[]( int index ) { return this->at( index ); }
```

Comment:
I'm fine with making both at and [] use refs& for both editing the object in the set and accessing if there are no objections. I figured they weren't ref due to the contents of the array being mostly smart_pointers, that we may want to create a local copy for when accessing (and because that's the way at() originally was). I can imagine very few circumstances where making them & would cause issues because the underlying set should usually still be in scope wheever they're used.

Reply:
That's a good point about the smart pointers. Lets leave it as return by value.